src/components/DatePicker.vue

```diff
@@ -504,9 +511,27 @@
           return D + ['th', 'st', 'nd', 'rd'][D % 10 > 3 ? 0 : (D - D % 10 !== 10) * D % 10];
         }
       };
+      let checkinDate = this.formatDate(new Date(this.checkIn) , 'YYYYMM');
+      let startDate = this.formatDate(new Date(this.startDate), 'YYYYMM');
+      let endDate = this.formatDate(this.getNextMonth(new Date(this.startDate)),'YYYYMM');
+      if(this.checkIn && (parseInt(endDate) < parseInt(checkinDate) || parseInt(startDate) < parseInt(checkinDate)) ){
+          let count = this.getMonthDiff(new Date(this.startDate), this.checkIn)
+          let nextMonth = new Date(this.startDate)
+           for(let i = 0; i < count; i++){
+            let tempNextMonth = this.getNextMonth(new Date(nextMonth))
+            this.createMonth(this.getNextMonth(new Date(nextMonth)))
+            nextMonth = tempNextMonth
+          }
+           if(this.checkOut && this.getMonthDiff(new Date(this.checkIn), new Date(this.checkOut)) > 0){
```

Comment:
We should probably move the `new Date` to the `getMonthDiff` method so we can avoid repeating it everytime we call this method.

Reply:
It is fixed